crates/turbopack-core/src/resolve/mod.rs

```diff
@@ -1709,29 +1709,12 @@ async fn resolve_internal_inline(
 async fn resolve_into_folder(
     package_path: Vc<FileSystemPath>,
     options: Vc<ResolveOptions>,
-    query: Vc<String>,
 ) -> Result<Vc<ResolveResult>> {
     let package_json_path = package_path.join("package.json".to_string());
     let options_value = options.await?;
+
     for resolve_into_package in options_value.into_package.iter() {
         match resolve_into_package {
-            ResolveIntoPackage::Default(req) => {
-                let str = "./".to_string()
-                    + &*normalize_path(req).ok_or_else(|| {
-                        anyhow!(
-                            "ResolveIntoPackage::Default can't be used with a request that \
-                             escapes the current directory"
-                        )
-                    })?;
-                let request = Request::parse(Value::new(str.into()));
-                return Ok(resolve_internal_inline(
-                    package_path,
-                    request.resolve().await?,
-                    options,
-                )
-                .await?
-                .with_request(".".to_string()));
-            }
             ResolveIntoPackage::MainField {
```

Comment:
It's a bit weird that the `into_package` is applied here. Do we need a separate resolve option `into_folder` next to `into_package` to set resolving into packages independent from packages?

Reply:
I don't think so as `MainField` would have to be in both?

`into_package` being a list doesn't really make sense to me though, it could probably just be a struct with 2 fields (also getting rid of the enum)